akka-cluster/src/main/scala/akka/cluster/Member.scala

```diff
@@ -271,17 +281,29 @@ object MemberStatus {
    */
   def removed: MemberStatus = Removed
 
+  /**
+   * Java API: retrieve the `ShuttingDown` status singleton
+   */
+  def shuttingDown: MemberStatus = PreparingForShutdown
+
+  /**
+   * Java API: retrieve the `ShutDown` status singleton
+   */
+  def shutDown: MemberStatus = ReadyForShutdown
+
   /**
    * INTERNAL API
    */
   private[cluster] val allowedTransitions: Map[MemberStatus, Set[MemberStatus]] =
     Map(
       Joining -> Set(WeaklyUp, Up, Leaving, Down, Removed),
-      WeaklyUp -> Set(Up, Leaving, Down, Removed),
-      Up -> Set(Leaving, Down, Removed),
+      WeaklyUp -> Set(Up, Leaving, Down, Removed, PreparingForShutdown),
```

Comment:
did you come to a conclusion about this? should WeaklyUp be treated as Joining instead, i.e. no transition to PreparingForShutdown?

Reply:
Yes I think so. I also thought that the leader won't move new nodes to Up once it has seen a preparing for shutdown

Reply:
sounds good